pym/bob/input.py

```diff
@@ -2012,13 +2019,23 @@ def prepare(self, pathFormatter, inputEnv, sandboxEnabled, inputStates, inputSan
                     env.update(sandbox.getEnvironment())
                     if dep.provideGlobal: depEnv.update(sandbox.getEnvironment())
 
+        # if a dependency is a direct dependency with a empty use-section and also provided by some other package use the buildResult
+        for p in providedPackages:
+            if p.getPackage().getName() in thisDeps:
+                # already have this dependency
+                if inResults.setdefault(p.getPackage().getName(), p) is p:
+                    # but not in the results list
+                    results.append(p)
+
         # filter indirect packages and add to result list
         tmp = indirectPackages
         indirectPackages = []
         for p in tmp:
             p2 = thisDeps.setdefault(p.getPackage().getName(), p)
             if p2 is p:
-                results.append(p)
+                p2 = inResults.setdefault(p.getPackage().getName(), p)
```

Comment:
This can only happen because of the above (wrong) hunk. The thing is that you may have to pick up the result if `p2 is not p`.